### DISCUSSION TITLE
Функция panelImg. Как поставить картинку на превью с другой таблицы?

### DISCUSSION THREAD

**Discussion by DmitriyGulevichS** [totumonline/totum-ru-issues-and-discussions#514] [Как это сделать?] [answered]
Добрый день! У меня таблица с товаром в которой я указываю количество комплектующих. Хотел сделать превью комплектующей с другой таблицы, где к каждой комплектующей я задам картинку. 

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/100049818/5bc0a3ce-2e0c-429b-8d7e-4d50a1e55914)

Вот к примеру 

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/100049818/dd45a5e4-8d9f-4081-b8a8-b7e082e82884)

вот что прописываю в коде форматирования 

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/100049818/50fcf94f-e1de-460d-9805-966cb85f9c24)

Что я делаю не так? 

PS. В документации прям пару слов написано про это, не понятно к сожалению

**Comment by ttm-alexeypolunin** [comment, marked as answer]
```
p1=: panelImg(img: $img[0][file])

img: select(table: 'foto'; field: 'kartinka'; where: 'id' = 1)
```

ток вместо единицы у вас должна быть ссылка на поле в этой таблице, в которой определено из какой строки вы берете

**Comment by DmitriyGulevichS** [comment]
Добрый день! Сделал аналогично коду выше, но так и не работает. Я подозреваю что ошибка прячется в вашем комментарии 

"ток вместо единицы у вас должна быть ссылка на поле в этой таблице, в которой определено из какой строки вы берете"

Тут не совсем понял что вы имете ввиду. Как достать именно ссылку?

**Reply by ttm-alexeypolunin** [reply]
Фото (файл) лежит на HDD и у него есть имя в системе linux. Информация об этом имени хранится в поле тип Файл, в которое был загружен этот файл. 

В вашем случае это таблица `foto`, поле `kartinka` строка `1`.

Если взять информацию функцией `select` из этого поля то там будет структура данных из нескольких строк и колонок (ключей). Несколько строк, тк в поле Файл может быть несколько файлов (если там мульти). Но если один, то там одна строка.

Информация  об имени файла в этой структуре лежит в колонке `file`. Таким образом взяв эту структуру, то можно получить имя файла `$struktura[0][file]` — `0` это первая строка `file` это взять из колонки `file`.

Теперь надо определить откуда мы берем структуру:

```
struktura: select(table: 'foto'; field: 'kartinka'; where: 'id' = $stroka)
```

Ключевой вопрос — из какой строки? Если она там всегда в первой строке этот винт лежит, то `$stroka = 1`

Если нет, то в какой строке...
